CMakeLists.txt

```diff
@@ -205,6 +205,18 @@ if(Boost_VERSION GREATER 107299)
   add_definitions(-DBOOST_BIND_GLOBAL_PLACEHOLDERS)
 endif()
 
+# Enable ccache automatically if available
+find_program(CCACHE_PROGRAM ccache)
+if(CCACHE_PROGRAM)
+  message(STATUS "Using ccache: ${CCACHE_PROGRAM}")
+  set(CMAKE_C_COMPILER_LAUNCHER
+      ${CCACHE_PROGRAM}
+      CACHE STRING "C compiler launcher" FORCE)
+  set(CMAKE_CXX_COMPILER_LAUNCHER
+      ${CCACHE_PROGRAM}
+      CACHE STRING "C++ compiler launcher" FORCE)
+endif()
+
```

Comment:
Never a good idea to force the use of `ccache`. It should be set using the CMAKE variable directly externally, not internally.

Reply:
Thanks for the feedback! I was indeed not sure on imposing this. I've reverted this.